File: android-tv-app/app/src/main/java/com/vugaenterprises/androidtv/data/model/AgeRating.kt
package com.vugaenterprises.androidtv.data.model

import com.google.gson.annotations.SerializedName

data class AgeRating(
    @SerializedName("age_limit_id")
    val id: Int,
    
    @SerializedName("name")
    val name: String,
    
    @SerializedName("min_age")
    val minAge: Int,
    
    @SerializedName("max_age")
    val maxAge: Int? = null,
    
    @SerializedName("description")
    val description: String? = null,
    
    @SerializedName("code")
    val code: String,
    
    @SerializedName("created_at")
    val createdAt: String? = null,
    
    @SerializedName("updated_at")
    val updatedAt: String? = null
) {
    // Helper to check if this rating is appropriate for kids
    val isKidsFriendly: Boolean
        get() = maxAge != null && maxAge <= 12
        
    // Helper to check if a given age can access this content
    fun canAccessByAge(age: Int?): Boolean {
        return age == null || age >= minAge
    }
    
    // Get display color based on age group
    val displayColor: String
        get() = when (code) {
            "AG_0_6" -> "#4CAF50" // Green
            "AG_7_12" -> "#8BC34A" // Light Green
            "AG_13_16" -> "#FF9800" // Orange
            "AG_17_18" -> "#F44336" // Red
            "AG_18_PLUS" -> "#9C27B0" // Purple
            else -> "#757575" // Gray
        }
}

data class AgeRatingResponse(
    @SerializedName("status")
    val status: Boolean,
    
    @SerializedName("message")
    val message: String,
    
    @SerializedName("age_ratings")
    val ageRatings: List<AgeRating>? = null
)

File: ios-mobile-app/Vuga/Models/AgeRating.swift
import Foundation

struct AgeRating: Codable, Identifiable, Equatable {
    let id: Int
    let name: String?  // Made optional to handle missing data
    let minAge: Int
    let maxAge: Int?
    let description: String?
    let code: String
    let createdAt: String?
    let updatedAt: String?
    
    enum CodingKeys: String, CodingKey {
        case id = "age_limit_id"
        case name
        case minAge = "min_age"
        case maxAge = "max_age"
        case description
        case code
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
    
    // Helper to check if this rating is appropriate for a given age
    func isAppropriateFor(age: Int?) -> Bool {
        guard let age = age else { return true }
        return age >= minAge
    }
    
    // Helper to check if this rating is appropriate for kids profiles
    var isKidsFriendly: Bool {
        // Kids profiles can access content for ages 12 and under
        if let maxAge = maxAge {
            return maxAge <= 12
        }
        return false
    }
    
    // Helper to get display color based on age group
    var displayColor: String {
        switch code {
        case "AG_0_6":
            return "#4CAF50" // Green
        case "AG_7_12":
            return "#8BC34A" // Light Green
        case "AG_13_16":
            return "#FF9800" // Orange
        case "AG_17_18":
            return "#F44336" // Red
        case "AG_18_PLUS":
            return "#9C27B0" // Purple
        default:
            return "#757575" // Gray
        }
    }
}

struct AgeRatingResponse: Codable {
    let status: Bool
    let message: String
    let ageRatings: [AgeRating]?
    
    enum CodingKeys: String, CodingKey {
        case status
        case message
        case ageRatings = "age_ratings"
    }
}

// Response for updating profile age settings
struct ProfileAgeUpdateResponse: Codable {
    let status: Bool
    let message: String
    let profile: Profile?
}